### PR TITLE
Fix some bugs in cloth generation

### PR DESCRIPTION
**Pull request by cgokmen**
This caps the max particle count to something we'll need to upper bound by later (due to GPU stuff), and also fixes an issue that causes each vertex to appear 6 times in the final mesh.

### CODE REVIEW

File: omnigibson/systems/micro_particle_system.py [hunk 1]
```diff
@@ -43,6 +44,7 @@
 
 # TODO: Tune these default values!
 # TODO (eric): figure out whether one offset can fit all
+m.MAX_CLOTH_PARTICLES = 20000
```
Comment (ChengshuLi) [inline omnigibson/systems/micro_particle_system.py:47]: is there a link to their doc that specify this magic number?

File: omnigibson/systems/micro_particle_system.py [hunk 2]
```diff
@@ -1540,44 +1542,58 @@ def clothify_mesh_prim(cls, mesh_prim, remesh=True, particle_distance=None):
...
+
+                avg_edge_percentage_mismatch = 1.0
+                # Loop re-meshing until average edge percentage is within error threshold or we reach the max number of tries
+                for _ in range(5):
+                    if avg_edge_percentage_mismatch <= m.CLOTH_REMESHING_ERROR_THRESHOLD:
+                        break
+
+                    ms.meshing_isotropic_explicit_remeshing(iterations=5, targetlen=pymeshlab.AbsoluteValue(particle_distance))
+                    avg_edge_percentage_mismatch = abs(1.0 - particle_distance / ms.get_geometric_measures()["avg_edge_length"])
+                else:
```
Comment (ChengshuLi) [inline omnigibson/systems/micro_particle_system.py:1573]: all the else statements in this function seem weird, can you double check? the logic is a bit hard to follow.

File: omnigibson/systems/micro_particle_system.py (hunk 1 above)
Comment (cgokmen) [reply omnigibson/systems/micro_particle_system.py:47]: The actual magic number is something around 22600 or something. It shows up in an error when you exceed it with GPU pipeline on, docs search doesnt point at anything. I left a bit of margin too.

File: omnigibson/systems/micro_particle_system.py (hunk 2 above)
Comment (cgokmen) [reply omnigibson/systems/micro_particle_system.py:1573]: just checked - the else statements run when the loop isnt broken out of via a `break` statement, which we do in the success cases. So in the failure cases we go into the elses.

**Comment by cgokmen** [comment]
I have tested this and confirmed the cloth demos still work.

File: omnigibson/systems/micro_particle_system.py [hunk 3]
```diff
@@ -1540,44 +1542,58 @@ def clothify_mesh_prim(cls, mesh_prim, remesh=True, particle_distance=None):
...
                     # Terminate anyways, but don't fail
                     log.warn("The generated cloth may not have evenly distributed particles.")
+
+                # Check if we have too many vertices
+                cm = ms.current_mesh()
+                if cm.vertex_number() > m.MAX_CLOTH_PARTICLES:
+                    # We have too many vertices, so we will re-mesh again
+                    particle_distance *= 1.47  # halve the number of vertices
```
Comment (cremebrule) [inline omnigibson/systems/micro_particle_system.py:1581]: what does 1.47 signify?

File: omnigibson/systems/micro_particle_system.py [hunk 4]
```diff
@@ -1540,44 +1542,58 @@ def clothify_mesh_prim(cls, mesh_prim, remesh=True, particle_distance=None):
...
+                else:
                     break
+            else:
+                raise ValueError("Could not remesh with less than MAX_CLOTH_PARTICLES vertices!")
```
Comment (cremebrule) [inline omnigibson/systems/micro_particle_system.py:1585]: maybe explicitly include the number here: `MAX_CLOTH_PARTICLES={m.MAX_CLOTH_PARTICLES}`

**Review by cremebrule — APPROVED** [review]
overall LGTM, a couple minor comments!

File: omnigibson/systems/micro_particle_system.py (hunk 3 above)
Comment (cgokmen) [reply omnigibson/systems/micro_particle_system.py:1581]: Bad attempt at approximating sqrt(2). Fixed.

File: omnigibson/systems/micro_particle_system.py (hunk 4 above)
Comment (cgokmen) [reply omnigibson/systems/micro_particle_system.py:1585]: Done.
